integration-tests/hibernate-validator/src/test/java/io/quarkus/it/hibernate/validator/HibernateValidatorFunctionalityTest.java

```diff
@@ -205,4 +206,114 @@ public void testInheritance() {
                 .statusCode(400)
                 .body(containsString("must not be null"));
     }
+
+    @Test
+    public void testRestEndPointValidationGroups_parameters() {
+        // PUT: input id must be null
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body("{\"id\": 1, \"name\": \"b\"}")
+                .when()
+                .put("/hibernate-validator/test/rest-end-point-validation-groups/")
+                .then()
+                .statusCode(400)
+                .body(containsString("must be null"));
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body("{\"name\": \"b\"}")
+                .when()
+                .put("/hibernate-validator/test/rest-end-point-validation-groups/")
+                .then()
+                .statusCode(200)
+                .body(containsString("passed"));
+
+        // POST: input id must not be null
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body("{\"name\": \"b\"}")
+                .when()
+                .post("/hibernate-validator/test/rest-end-point-validation-groups/")
+                .then()
+                .statusCode(400)
+                .body(containsString("must not be null"));
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body("{\"id\": 1, \"name\": \"b\"}")
+                .when()
+                .post("/hibernate-validator/test/rest-end-point-validation-groups/")
+                .then()
+                .statusCode(200)
+                .body(containsString("passed"));
+
+        // Also check that constraints using the default group still work
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body("{}")
+                .when()
+                .put("/hibernate-validator/test/rest-end-point-validation-groups/")
+                .then()
+                .statusCode(400)
+                .body(containsString("must not be null"));
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body("{\"id\":1}")
+                .when()
+                .post("/hibernate-validator/test/rest-end-point-validation-groups/")
+                .then()
+                .statusCode(400)
+                .body(containsString("must not be null"));
+    }
+
+    @Test
+    public void testRestEndPointValidationGroups_result() {
+        // GET: deleted must be false
+        RestAssured.given()
+                .param("simulateDeleted", true)
+                .when()
+                .get("/hibernate-validator/test/rest-end-point-validation-groups/1/")
+                .then()
+                .statusCode(500)
```

Comment:
I suppose `ConstraintViolationExceptions` always result in 500 for `DELETE`, even without validation groups?
I find this a bit odd since I would have expected 400, just like for `PUT` and `POST`?

Reply:
It's not related to the HTTP method. The validation is performed on the result in this case, which is why a validation failure leads to a HTTP status 500 instead of 400.

Makes sense, since an incorrect result is to blame on the server, not on the caller.

Reply:
Ah yes, makes perfect sense. Thanks for explaining!